src/app/settings/Tokens.tsx

```diff
@@ -0,0 +1,293 @@
+import { PlusIcon } from '@heroicons/react/24/outline';
+import { format, parseISO } from 'date-fns';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useLoaderData } from 'react-router-dom';
+import EmptyState from '~/components/EmptyState';
+import Button from '~/components/forms/Button';
+import Modal from '~/components/Modal';
+import Slideover from '~/components/Slideover';
+import DeleteTokenPanel from '~/components/tokens/DeleteTokenPanel';
+import ShowTokenPanel from '~/components/tokens/ShowTokenPanel';
+import { listTokens } from '~/data/api';
+import { useError } from '~/data/hooks/error';
+import {
+  IAuthToken,
+  IAuthTokenList,
+  IAuthTokenSecret
+} from '~/types/auth/Token';
+import TokenForm from './TokenForm';
+
+export async function tokenLoader(): Promise<IAuthTokenList> {
+  return listTokens();
+}
+
+export default function Tokens() {
+  // const checkbox = useRef();
+  const data = useLoaderData() as IAuthTokenList;
+  const [tokens, setTokens] = useState<IAuthToken[]>(data.authentications);
+
+  const [tokensVersion, setTokensVersion] = useState(0);
+
+  const { setError, clearError } = useError();
+
+  const fetchTokens = useCallback(() => {
+    listTokens()
+      .then((data) => {
+        setTokens(data.authentications);
+        clearError();
+      })
+      .catch((err) => {
+        setError(err);
+      });
+  }, [clearError, setError]);
+
+  const incrementTokensVersion = () => {
+    setTokensVersion(tokensVersion + 1);
+  };
+
+  useEffect(() => {
+    fetchTokens();
+  }, [tokensVersion, fetchTokens]);
+
+  // const [checked, setChecked] = useState(false);
+  // const [indeterminate, setIndeterminate] = useState(false);
+  // const [selectedTokens, setSelectedTokens] = useState<IAuthenticationToken[]>(
+  //   []
+  // );
+
+  const [createdToken, setCreatedToken] = useState<IAuthTokenSecret | null>(
+    null
+  );
+  const [showCreatedTokenModal, setShowCreatedTokenModal] = useState(false);
+
+  const [showTokenForm, setShowTokenForm] = useState<boolean>(false);
+
+  const [showDeleteTokenModal, setShowDeleteTokenModal] =
+    useState<boolean>(false);
+  const [deletingToken, setDeletingToken] = useState<IAuthToken | null>(null);
+
+  // useLayoutEffect(() => {
```

Comment:
Lots of commented out code?

Reply:
yeah, this is for when we want to enable deleting multiple via checkboxes. This is coming in a future implementation, so I didn't want to remove the code as we'll use it soon